pkg/apis/appstacks/v1beta1/runtimecomponent_types.go

```diff
@@ -75,6 +75,10 @@ type RuntimeComponentService struct {
 	// +kubebuilder:validation:Minimum=1
 	TargetPort *int32 `json:"targetPort,omitempty"`
 
+	// +kubebuilder:validation:Maximum=65535
+	// +kubebuilder:validation:Minimum=1
+	NodePort int32 `json:"nodePort,omitempty"`
```

Comment:
I think setting this constraints will not allow user switch back to ClusterIP type as `nodePort` cannot be null in this case
Also once set to specific value, there is no way to make it nil again `0` or `nil` values will be not accepted. So there is no way to to switch back to auto-generated port number.
So I think the 0 should be allowed